Sources/MongoSwift/ChangeStreamOptions.swift

```diff
@@ -0,0 +1,101 @@
+/// Describes the modes for configuring the fullDocument field of a
+/// `ChangeStreamDocument`.
+public enum FullDocument: RawRepresentable, Codable {
+    /// The change stream document will include both a delta describing the
+    /// changes to the document, as well as a copy of the entire document that
+    /// was changed from some time after the change occurred.
+    case updateLookup
+    /// For an unknown value. For forwards compatibility, no error will be
+    /// thrown when an unknown value is provided.
+    case other(String)
+
+    public var rawValue: String {
+        switch self {
+        case .updateLookup:
+            return "updateLookup"
+        case .other(let v):
+            return v
+        }
+    }
+
+    public init?(rawValue: String) {
+        switch rawValue {
+        case "updateLookup":
+            self = .updateLookup
+        default:
+            self = .other(rawValue)
+        }
+    }
+}
+
+/// Options to use when creating a `ChangeStream`.
+public struct ChangeStreamOptions: Codable {
+    /** Indicates how the `fullDocument` field of a `ChangeStreamDocument` should
+     * be filled out by the server.
+     * By default (indicated by a nil value for this option), the fullDocument field
+     * in the change stream document will always be present in the case of 'insert'
+     * and 'replace' operations (containing the document being inserted) and will be
+     * nil for all other operations.
+     */
+    public let fullDocument: FullDocument?
+
+    /// Specifies the logical starting point for the new change stream.
+    public let resumeAfter: Document?
+
+    /// The maximum amount of time in milliseconds for the server to wait on new documents
+    // to satisfy a change stream query. Uses the server default timeout when omitted.
+    public let maxAwaitTimeMS: Int64?
+
+    /**
+     * The number of documents to return per batch.
+     * This option is sent only if the caller explicitly provides a value. The
+     * default is to not send a value.
+     * - SeeAlso: https://docs.mongodb.com/manual/reference/command/aggregate
+     */
+    public let batchSize: Int32?
+
+    /**
+     * Specifies a collation.
+     * This option is sent only if the caller explicitly provides a value. The
+     * default is to not send a value.
+     * - SeeAlso: https://docs.mongodb.com/manual/reference/command/aggregate
+     */
+    public let collation: Document?
+
+    /**
+     * The change stream will only provide changes that occurred at or after
+     * the specified timestamp. Any command run against the server will return
+     * an operation time that can be used here.
+     * - SeeAlso: https://docs.mongodb.com/manual/reference/method/db.runCommand/
+     */
+    public let startAtOperationTime: Timestamp?
+
+    /**
+     * Similar to `resumeAfter`, this option takes a resume token and starts a
```

Comment:
This docstring is really good / informative. If you could update the `resumeAfter` one to look like this, that would be great.

One interesting thing to note: this is a 4.2 feature, so libmongoc might not have implemented it fully yet. If so, we may have to keep it internal until libmongoc does.

Reply:
@patrickfreed I see `startAfter` being used in mongoc here: https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/src/mongoc/mongoc-change-stream.c#L122

Reply:
that's the master branch of libmongoc, which contains unreleased code. if you look at the [r1.14 branch version of the file](https://github.com/mongodb/mongo-c-driver/blob/r1.14/src/libmongoc/src/mongoc/mongoc-change-stream.c), which is what is in the 1.14 release, it's not there.
so I don't think we can support `startAfter` yet unfortunately.

Reply:
so I think we need to make this property internal and add a comment under the docstring like: "TODO: make this public when support is added for 4.2 change stream features". and then remove the `startAfter` param from the options init and remove mention of it from elsewhere in the file

Reply:
Swift still wants the `startAfter` property to be initialized. Should we just remove this from the code for now (or store some value/comment it all out?) ?

Reply:
Yeah, that should be fine. Also can you file a ticket so that we remember to do the work and update the TODO with the ticket number?